src/riak_object.erl

```diff
@@ -1064,11 +1073,34 @@ fold_meta_to_bin(Key, Value, {{_Vt,_Del,_Lm}=Elems,RestBin}) ->
     MetaBin = <<KeyLen:32/integer, KeyBin/binary, ValueLen:32/integer, ValueBin/binary>>,
     {Elems, <<RestBin/binary, MetaBin/binary>>}.
 
+%% @doc Encode the contents of a riak object, either using
+%% term_to_binary (if Enc == erlang), or msgpack encoding (if Enc ==
+%% msgpack)
+encode(Bin, Enc) when Enc == erlang ->
```

Comment:
Instead of using a guard, is it more "Erlangish" to pattern match?

`encode(Bin, erlang)` and `encode(Bin, msgpack)`